### PR TITLE
Wdio up rb

### PR DESCRIPTION
**Pull request by alisman**
Fix cBioPortal/cbioportal# (see https://help.github.com/en/articles/closing-issues-using-keywords)

Describe changes proposed in this pull request:
- a
- b

## Checks
- [ ] Has tests or has a separate issue that describes the types of test that should be created. If no test is included it should explicitly be mentioned in the PR why there is no test.
- [ ] The commit log is comprehensible. It follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Is this PR adding logic based on one or more **clinical** attributes? If yes, please make sure validation for this attribute is also present in the data validation / data loading layers (in backend repo) and documented in [File-Formats Clinical data section](https://github.com/cBioPortal/cbioportal/blob/master/docs/File-Formats.md#clinical-data)!

## Any screenshots or GIFs?
If this is a new visual feature please add a before/after screenshot or gif
here with e.g. [Giphy CAPTURE](https://giphy.com/apps/giphycapture) or [Peek](https://github.com/phw/peek)

## Notify reviewers
Read our [Pull request merging
policy](../CONTRIBUTING.md#pull-request-merging-policy). It can help to figure out who worked on the
file before you. Please use `git blame <filename>` to determine that
and notify them either through slack or by assigning them as a reviewer on the PR

### CODE REVIEW

File: scripts/e2e.sh [hunk 1]
```diff
@@ -0,0 +1,71 @@
+MY_PATH="`dirname \"$0\"`"              # relative
```
Comment (alisman) [inline scripts/e2e.sh:1]: @inodb @pvannierop this is script i have written to hopefully make booting up localdb mostly single command.

File: scripts/e2e.sh [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        [Yy]* ) SPIN_UP=true;break;;
+        [Nn]* ) break;;
+        * ) break;;
+    esac
+done
+
+echo $SPIN_UP;
+
+
+export PORTAL_SOURCE_DIR=$PWD;
```
Comment (alisman) [inline scripts/e2e.sh:39]: for all of these exports, what I plan to do is only set them to these defaults if the are NOT already set.  that way people can configure them in custom fashion as necessary.  most people will just rely on these defaults

Comment (inodb) [reply scripts/e2e.sh:39]: you can use:

```
export PORTAL_SOURCE_DIR ="${PORTAL_SOURCE_DIR:-${PWD}}"
```

File: README.md [hunk 3]
```diff
@@ -190,87 +189,49 @@ Files for the local database e2e-tests are located in the `./end-to-end-test/loc
...
-cd ~/cbioportal-docker-compose
-docker-compose run --rm cbioportal sh -c '
-    cd /cbioportal/core/src/main/scripts/ \
-    && ./importGenePanel.pl --data /cbioportal/core/src/test/scripts/test_data/study_es_0/data_gene_panel_testpanel1.txt \
-    && ./importGenePanel.pl --data /cbioportal/core/src/test/scripts/test_data/study_es_0/data_gene_panel_testpanel2.txt \
-    && ./importGenesetData.pl --data /cbioportal/core/src/test/resources/genesets/study_es_0_genesets.gmt --new-version msigdb_6.1 \
-    && ./importGenesetHierarchy.pl --data /cbioportal/core/src/test/resources/genesets/study_es_0_tree.yaml'
+export BRANCH_ENV=custom
+yarn install --frozen-lockfile // only necessary first time
+yarn buildDLL:dev // only necessary first tiem
```
Comment (inodb) [inline README.md:202]: prolly good to use `#` here for comment, cause then you can copy+paste them (`//` is not a comment in bash)

File: scripts/e2e.sh (hunk 2 above)
Comment (alisman) [reply scripts/e2e.sh:39]: oooh.  so nice.

File: README.md [hunk 4]
```diff
@@ -190,87 +189,49 @@ Files for the local database e2e-tests are located in the `./end-to-end-test/loc
...
 
-Restart cBioPortal
-```
-docker-compose restart cbioportal
+3. Install dev dependencies:
+```bash
+cd end-to-end-test
+yarn
 ```
+5. In a second terminal at project root, spinup the backend (api) instance:
```
Comment (inodb) [inline README.md:211]: No step 4 :)?

File: end-to-end-test/local/docker_compose/docker/Dockerfile.screenshottest [hunk 5]
```diff
@@ -1,10 +1,11 @@
-FROM circleci/node:8.12.0-browsers
+FROM circleci/node:15.2.1-browsers
 MAINTAINER Pim van Nierop <pim@thehyve.nl>
 
 LABEL Description="End-2-end test image for cBioPortal frontend"
 
-USER root
+#echo INSTALL cavas deps [necessary for visual regression framework]
+#RUN apt-get install build-essential libcairo2-dev libpango1.0-dev libjpeg-dev libgif-dev librsvg2-dev
```
Comment (inodb) [inline end-to-end-test/local/docker_compose/docker/Dockerfile.screenshottest:7]: this prolly doesn't work anymore after commenting out?

**Review by inodb — COMMENTED** [review]
Thanks for doing this!!

I added some comments, I think it's mostly clean up stuff and making sure the existing tests continue working. Also some description for the PR would be good

I tried following the steps but im still having some issue with the mysql database. It doesn't seem to actually load the seed data somehow, so getting errors like:

```
(1146, "Table 'cbioportal.mutation_event' doesn't exist")
```

File: .circleci/config.yml [hunk 6]
```diff
@@ -6,26 +6,25 @@ defaults: &defaults
...
 run_e2e_and_save_artifacts: &run_e2e_and_save_artifacts
   steps:
-    - run:
-        name: "Install yarn at specific version"
-        command:
-          sudo npm install --global yarn@1.22.4
+#    - run:
+#        name: "Install yarn at specific version"
+#        command:
+#          sudo npm install --global yarn@1.22.4
```
[the remark this answers is not in the document]
Comment (alisman) [reply .circleci/config.yml:16]: the new image has yarn installed already

File: README.md [hunk 7]
```diff
@@ -158,8 +158,7 @@ webdriver-manager start
 
 In a second terminal, run the frontend
 ```bash
-# set the environment variables for your branch
-export BRANCH_ENV=master # or rc if branching from rc
```
[the remark this answers is not in the document]
Comment (alisman) [reply README.md:162]: yeah, but only necessary if you need to change it from master

File: scripts/e2e.sh (hunk 1 above)
Comment (alisman) [reply scripts/e2e.sh:1]: thumb sup

File: .circleci/config.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (inodb) [reply .circleci/config.yml:16]: gotcha - maybe better to remove the lines in that case? Some info about this in the PR description would be nice

**Pull request closed without merging**